src/Razor/test/Microsoft.VisualStudio.Razor.IntegrationTests/CodeFoldingTests.cs

```diff
@@ -0,0 +1,186 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Host;
+using Microsoft.VisualStudio.Text;
+using Microsoft.VisualStudio.Text.Editor;
+using Microsoft.VisualStudio.Text.Outlining;
+using Xunit;
+
+namespace Microsoft.VisualStudio.Razor.IntegrationTests
+{
+    public class CodeFoldingTests : AbstractRazorEditorTest
```

Comment:
Still adding more tests, but this is the rough infrastructure to add them

Reply:
If you feel it'd be easier we can check this in as-is and revisit tests too. Don't want them to hold you up 😄